http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -119,26 +118,36 @@ case class ImageAutoDetectionException(traceId: TraceId, image: ContainerImage)
     extends LeoException(s"${traceId} | Unable to auto-detect tool for ${image.registry} image ${image.imageUrl}",
                          StatusCodes.Conflict)
 
-class LeonardoService(protected val dataprocConfig: DataprocConfig,
-                      protected val imageConfig: ImageConfig,
-                      protected val welderDao: WelderDAO[IO],
-                      protected val clusterDefaultsConfig: ClusterDefaultsConfig,
-                      protected val proxyConfig: ProxyConfig,
-                      protected val swaggerConfig: SwaggerConfig,
-                      protected val autoFreezeConfig: AutoFreezeConfig,
-                      protected val welderConfig: WelderConfig,
-                      protected val petGoogleStorageDAO: String => GoogleStorageDAO,
-                      protected val authProvider: LeoAuthProvider[IO],
-                      protected val serviceAccountProvider: ServiceAccountProvider[IO],
-                      protected val bucketHelper: BucketHelper,
-                      protected val clusterHelper: ClusterHelper,
-                      protected val dockerDAO: DockerDAO[IO])(implicit val executionContext: ExecutionContext,
-                                                              implicit override val system: ActorSystem,
-                                                              log: Logger[IO],
-                                                              cs: ContextShift[IO],
-                                                              metrics: NewRelicMetrics[IO],
-                                                              dbRef: DbReference[IO],
-                                                              timer: Timer[IO])
+case class UpdateResult(hasUpdateSucceded: Boolean, shouldFollowup: Boolean, followupAction: UpdateTransition)
+
+sealed trait UpdateTransition
+case class StopStartTransition(updateConfig: MachineConfig) extends UpdateTransition
```

Comment:
move all sub classes to an object `UpdateTransition`...that way it's clearer what all the sub classes are